### ISSUE TITLE
bug(MatMenu): When opening a submenu and hovering over any root menu entry, the menu can't be closed using ESC

### ISSUE THREAD

**Issue by localh0rzd** [angular/components#24993]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

The keyboard navigation of `<mat-menu>` submenus doesn't work as expected when combining it with mouse navigation. If a submenu is opened, closing the menu via ESC is only possible as long as the submenu isn't closed (e.g. by hovering over another menu item in the root menu).

### Reproduction

Stackblitz: https://stackblitz.com/edit/components-issue-hjkjwg?file=src/app/example-component.html

Steps to reproduce:
1. Open the menu
2. Hover over Item 2 --> Submenu opens
3. Hover over Item 3 --> Submenu closes
4. Try hitting ESC --> Menu doesn't close

Same applies to the Subsubmenu in regards of any parent menu

### Expected Behavior

Menu should close when ESC is pressed

### Actual Behavior

Menu doesn't close. ESC works as long as the last opened submenu is pressed while it is still visible

### Environment

- Angular: 13.3.0
- CDK/Material: 13.3.0
- Browser(s): Chrome 102
- Operating System (e.g. Windows, macOS, Ubuntu): Windows 10

**Comment by crisbeto** [comment]
We are closing this issue, because we've determined, using a semi-automated process, that it is no longer relevant, has low engagement, or it does not have enough information to proceed. Feel free to comment if you think the issue should be reopened, or open a new issue with the latest version of our code instead.
